test/test-server/hub_handlers.go

```diff
@@ -0,0 +1,129 @@
+package testserver
+
+import (
+	"crypto/md5"
+	"crypto/sha1"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/require"
+
+	"github.com/confluentinc/cli/internal/pkg/cpstructs"
+	"github.com/confluentinc/cli/internal/pkg/utils"
+)
+
+const archivePath = "test/fixtures/input/connect/test-plugin.zip"
+
+func newManifest() *cpstructs.Manifest {
+	return &cpstructs.Manifest{
+		Name:    "integration-test-plugin",
+		Title:   "Integration Test Plugin",
+		Version: "0.1.0",
+		Owner: cpstructs.Owner{
+			Username: "confluentinc",
+			Name:     "Confluent, Inc.",
+		},
+		Licenses: []cpstructs.License{
+			{
+				Name: "Apache License 2.0",
+				Url:  "https://www.apache.org/licenses/LICENSE-2.0",
+			},
+		},
+	}
+}
+
+// Handler for: "/api/plugins/{owner}/{id}"
```

Comment:
Has this team written an OpenAPI spec, or have you asked them to? It would be awesome to have an SDK for these endpoints. As of right now, I don't think `--unsafe-trace` will work.

Reply:
I'll ask.

Unsafe trace works; I put support for it in the functions in `internal/pkg/hub/manifest.go`. Here's an example (I cut out the response body since it's pretty large):
```
connect plugin install confluentinc/kafka-connect-datagen:latest --dry-run --unsafe-trace
2023-06-06T12:10:46.073+0800 [TRACE] 
GET /api/plugins/confluentinc/kafka-connect-datagen HTTP/1.1
Host: api.hub.confluent.io
User-Agent: Confluent-CLI/v3.16.0-SNAPSHOT-cdc7f720e (https://confluent.io; support@confluent.io)
Accept-Encoding: gzip



2023-06-06T12:10:46.073+0800 [DEBUG] performing request%!(EXTRA string=method, string=GET, string=url, *url.URL=https://api.hub.confluent.io/api/plugins/confluentinc/kafka-connect-datagen)
2023-06-06T12:10:47.028+0800 [TRACE] 
HTTP/1.1 200 OK
Content-Length: 2762
Access-Control-Allow-Origin: null
Cache-Control: no-transform, s-maxage=26
Connection: keep-alive
Content-Type: application/vnd.confluent.v1+json
Date: Tue, 06 Jun 2023 04:10:47 GMT
Etag: "9c90bbe00bf0b8bf2cba64d8c5e8bd7ffa4a9e6369a9869fc3679017a3279480"
Server: nginx
```

I did notice while testing this though that I didn't set the user agent (it's fixed in this output); I'll push that momentarily.

Reply:
There doesn't appear to be an sdk but there's an openapi doc, I'll link it to you in slack if you're interested.